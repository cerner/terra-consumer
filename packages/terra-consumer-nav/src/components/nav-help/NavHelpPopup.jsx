import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import Button from 'terra-button';
import IconClose from 'terra-icon/lib/icon/IconClose';
import TerraPopup from 'terra-popup';
import SafeHtml from '../safe-html/SafeHtml';
import styles from './NavHelpPopup.scss';

const cx = classNames.bind(styles);

const propTypes = {
  /**
   * Title to be rendered on top of the popup content
   */
  title: PropTypes.string,
  /**
   * Boolean value to hide/display popup header with title and close button
   */
  hasHeader: PropTypes.bool,
  /**
   * Content to be displayed in a dialog
   */
  popupContent: PropTypes.element,
  /**
   * Boolean value to render the popup
   */
  isOpen: PropTypes.bool,
  /**
   * Required callback function for use by parent component to update state.
   */
  closePopup: PropTypes.func,
};

const defaultProps = {
  hasHeader: false,
  isOpen: false,
  closePopup: null,
};

const NavHelpPopup = ({
 title,
 hasHeader,
 popupContent,
 isOpen,
 closePopup,
 ...customProps
}) => {
  const popupHeader = (hasHeader &&
    <div className={cx('popup-title')}>
      <SafeHtml text={title} />
      <Button
        className={cx('close-button')}
        onClick={closePopup}
        variant={'link'}
        icon={<IconClose />}
        isCompact
      />
    </div>
  );

  return (
    <TerraPopup
      {...customProps}
      isOpen={isOpen}
      onRequestClose={closePopup}
      isHeaderDisabled
    >
      <div>
        {popupHeader}
        <div className={cx('popup-body')}>
          {popupContent}
        </div>
      </div>
    </TerraPopup>
  );
};

NavHelpPopup.propTypes = propTypes;
NavHelpPopup.defaultProps = defaultProps;

export default NavHelpPopup;
