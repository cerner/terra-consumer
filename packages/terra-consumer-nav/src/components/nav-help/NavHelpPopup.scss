:local {
  .popup-title {
    background: var(--terra-consumer--popup-title-background, #64696c);
    color: var(--terra-consumer--popup-title-color, #fff);
    font-size: 20px;
    line-height: 30px;
    padding: 5px 10px;
    width: 100%;
  }

  .close-button-container {
    border: 0;

  }

  .close-button {
    color: var(--terra-consumer--popup-close-button-color, #fff) !important;
    float: right !important;
    font-size: 16px;
    margin: 0 !important;
    padding: 0 !important;

    &:hover {
      color: var(--terra-consumer--popup-close-button-hover-color, #ccc) !important;
    }

    &:focus {
      color: var(--terra-consumer--popup-close-button-hover-color, #777) !important;
    }
  }

  .popup-body {
    height: 200px;
    overflow: scroll;
  }
}
