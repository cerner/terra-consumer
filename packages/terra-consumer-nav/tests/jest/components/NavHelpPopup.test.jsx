import React from 'react';
import NavHelpPopup from '../../../src/components/nav-help/NavHelpPopup';


const popupContent = <div> This is a test content</div>;
const mockFn = jest.fn();

describe('Popup', () => {
  // Snapshot Tests
  it('should render a pop with title,close button and content', () => {
    const wrapper = shallow(<NavHelpPopup
      title="Title"
      hasHeader
      targetRef={mockFn}
      closePopup={mockFn}
      contentWidth="320"
      contentHeight="240"
      contentAttachment="middle center"
      popupContent={popupContent}
      rowCount={1}
    />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render a popup without header(title & close button)', () => {
    const wrapper = shallow(<NavHelpPopup
      title="Title"
      hasHeader={false}
      targetRef={mockFn}
      closePopup={mockFn}
      contentWidth="320"
      contentHeight="240"
      contentAttachment="middle center"
      popupContent={popupContent}
      rowCount={1}
    />);
    expect(wrapper).toMatchSnapshot();
  });
});
