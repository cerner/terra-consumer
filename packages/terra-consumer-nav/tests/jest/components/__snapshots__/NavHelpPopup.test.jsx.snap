// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popup should render a pop with title,close button and content 1`] = `
<Popup
  attachmentBehavior="auto"
  boundingRef={null}
  classNameArrow={null}
  classNameContent={null}
  classNameOverlay={null}
  contentAttachment="middle center"
  contentHeight="auto"
  contentWidth="320"
  isArrowDisplayed={false}
  isContentFocusDisabled={false}
  isHeaderDisabled={true}
  isOpen={false}
  onRequestClose={[Function]}
  targetRef={[Function]}
>
  <div>
    <div
      className="popup-title"
    >
      <SafeHtml
        text="Title"
      />
      <Button
        className="close-button"
        icon={
          <IconClose
            data-name="Layer 1"
            viewBox="0 0 48 48"
            xmlns="http://www.w3.org/2000/svg"
          />
        }
        isBlock={false}
        isCompact={true}
        isDisabled={false}
        isReversed={false}
        onClick={[Function]}
        type="button"
        variant="link"
      />
    </div>
    <div>
      <div>
         This is a test content
      </div>
    </div>
  </div>
</Popup>
`;

exports[`Popup should render a popup without header(title & close button) 1`] = `
<Popup
  attachmentBehavior="auto"
  boundingRef={null}
  classNameArrow={null}
  classNameContent={null}
  classNameOverlay={null}
  contentAttachment="middle center"
  contentHeight="auto"
  contentWidth="320"
  isArrowDisplayed={false}
  isContentFocusDisabled={false}
  isHeaderDisabled={true}
  isOpen={false}
  onRequestClose={[Function]}
  targetRef={[Function]}
>
  <div>
    <div>
      <div>
         This is a test content
      </div>
    </div>
  </div>
</Popup>
`;
